### PR TITLE
Fix failure setting optimize option (#480)

### LINKED ISSUE #480: Failure when setting optimize option
This is because pyjulia uses ArgumentsParser to parse --home & --sysimage, but it get the optimize value as an int instead of a string.

I'll submit a PR

### PR DESCRIPTION
**Pull request by dpinol**
Fixes #480

### CODE REVIEW

**Comment by mkitti** [comment]
We would need to get these tests passing

**Comment by mkitti** [comment]
@MilesCranmer Clone the branch, get the tests to pass, and submit a new PR pinging me. Then I will merge it.

**Comment by dpinol** [comment]
logs have expired. Can sb rerun the tests?

**Comment by mkitti** [comment]
Considering rebasing

**Comment by dpinol** [comment]
@mkitti done

**Comment by mkitti** [comment]
Hmm, that did not seem to fix testing.

**Comment by MilesCranmer** [comment]
It seems unrelated to this PR - the issue seems to be because the CI is trying to download a pip package from an HTTP URL instead of HTTPS, and urllib no longer allows non-SSL requests:
```
      Downloading http://pypi.python.org/packages/source/d/distribute/distribute-0.6.27.tar.gz
```
I think that's why the Python 2.7 version works - since the old urllib hasn't deprecated non-SSL.

**Comment by MilesCranmer** [comment]
FYI @dpinol once #508 is merged this should work

**Comment by dpinol** [comment]
@MilesCranmer rebased. thanks

**Comment by MilesCranmer** [comment]
@mkitti any idea how to change the actions for “Required statuses must pass before merging”? It seems like they are for a different CI naming scheme maybe.

**Review by MilesCranmer — APPROVED** [review]
Looks good to me.

File: src/julia/tests/test_juliaoptions.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
     (dict(compiled_modules="no"), ["--compiled-modules", "no"]),
     (dict(depwarn="error"), ["--depwarn", "error"]),
     (dict(sysimage="PATH"), ["--sysimage", "PATH"]),
-    (dict(bindir="PATH"), ["--home", "PATH"]),
```
Comment (mkitti) [inline src/julia/tests/test_juliaoptions.py:14]: What happened to `bindir`?

**Comment by mkitti** [comment]
Status checks should be functional now

File: src/julia/tests/test_juliaoptions.py (hunk 1 above)
Comment (mkitti) [reply src/julia/tests/test_juliaoptions.py:14]: Other than this merge when ready.

Comment (MilesCranmer) [reply src/julia/tests/test_juliaoptions.py:14]: Is bindir an option in Julia anymore? I don't see it in the `--help` menu.

**Comment by MilesCranmer** [comment]
Could this one be merged?

File: src/julia/tests/test_juliaoptions.py (hunk 1 above)
Comment (mkitti) [reply src/julia/tests/test_juliaoptions.py:14]: https://github.com/JuliaLang/julia/blob/master/src/jloptions.c

Comment (MilesCranmer) [reply src/julia/tests/test_juliaoptions.py:14]: Not sure, I can't even use it locally:

```bash
julia --bindir=$PATH
# ERROR: unknown option `--bindir
```

Comment (MilesCranmer) [reply src/julia/tests/test_juliaoptions.py:14]: I think it's replaced by `--home`?

Comment (MilesCranmer) [reply src/julia/tests/test_juliaoptions.py:14]: Oh yeah, there it is:

```python
    def cli_argument_name(self):
        name = {"bindir": "home"}.get(self.name, self.name)
```

So this should not be removed @dpinol

File: src/julia/tests/test_juliaoptions.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
...
+    (dict(optimize=3), ["--optimize", "3"]),
```
Comment (mkitti) [inline src/julia/tests/test_juliaoptions.py:14]:
```suggestion
    (dict(bindir="PATH"), ["--home", "PATH"]),
    (dict(optimize=3), ["--optimize", "3"]),
```

File: src/julia/tests/test_juliaoptions.py (hunk 1 above)
Comment (mkitti) [reply src/julia/tests/test_juliaoptions.py:14]: I readded it.

**Comment by MilesCranmer** [comment]
Awesome! 🚀

**Pull request merged**